pfcpiface/pfcpiface.go

```diff
@@ -85,6 +88,18 @@ func pfcpifaceMainLoop(upf *upf, n3ip string, sourceIP string) {
 			continue
 		}
 
+		// if sourceIP is not set, fetch it from the msg header
+		if sourceIP == "0.0.0.0" {
+			header, err := ipv4.ParseHeader(buf[:n])
+			if err == nil {
+				log.Println("N4 Src IP gets resolved to:", header.Dst.String())
+				sourceIP = header.Dst.String()
+			} else {
+				log.Println("Failed to parse message header. Skipping...")
+				continue
+			}
```

Comment:
Use the net.IP from the IE instead and use the function you have to figure out outbound IP.
remove "golang.org/x/net/ipv4"

Reply:
Found an easier way.